runtime/scripts/question.js

```diff
@@ -1219,6 +1231,7 @@ Question.prototype = /** @lends Numbas.Question.prototype */
         if(this.exam && this.exam.adviceType == 'threshold' && 100*this.score/this.marks < this.adviceThreshold ) {
             this.getAdvice();
         }
+        this.events.trigger('submit');
```

Comment:
I think an event called `submit` at the top of the function, and one here called `submitted` would make sense